File: src/lib/scala/Slate.Common/src/main/scala/slate/common/InputArgs.scala
/**
  * <slate_header>
  * author: Kishore Reddy
  * url: https://github.com/kishorereddy/scala-slate
  * copyright: 2016 Kishore Reddy
  * license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  * desc: a scala micro-framework
  * usage: Please refer to license on github for more info.
  * </slate_header>
  */
package slate.common


class InputArgs(protected val _map:Map[String,Any],
                private val _decryptor:Option[(String) => String] = None) extends Inputs {

  override def getString   (key: String) : String   = InputFuncs.decrypt(_map(key).toString, _decryptor)
  override def getBool     (key: String) : Boolean  = _map(key).toString.toBoolean
  override def getInt      (key: String) : Int      = _map(key).toString.toInt
  override def getLong     (key: String) : Long     = _map(key).toString.toLong
  override def getDouble   (key: String) : Double   = _map(key).toString.toDouble
  override def getFloat    (key: String) : Float    = _map(key).toString.toFloat
  override def getDate     (key: String) : DateTime = {
    _map(key) match {
      case d:DateTime => d
      case s:String   => InputFuncs.convertDate(s)
      case n:Long     => DateTime.parseNumericDate12(n.toString)
    }
  }

  override def get(key: String) : Option[Any]             = if (_map.contains(key)) Option(_map(key)) else None
  override def getObject(key: String): Option[AnyRef]     = if (_map.contains(key)) Option(_map(key).asInstanceOf[AnyRef]) else None
  override def containsKey(key: String): Boolean          = _map.contains(key)
  override def size(): Int                                = _map.size
}

File: src/lib/kotlin/slatekit-common/src/main/kotlin/slatekit/common/InputArgs.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package slatekit.common

import java.time.LocalDate
import java.time.LocalDateTime
import java.time.LocalTime
import java.time.ZonedDateTime

/**
 * Created by kishorereddy on 5/25/17.
 */
class InputArgs(val _map: Map<String, Any>,
                private val _decryptor: ((String) -> String)? = null) : Inputs {

    override val raw:Any =_map

    override fun getString(key: String): String = InputFuncs.decrypt(_map[key].toString(), _decryptor)
    override fun getBool(key: String): Boolean = Conversions.toBool(_map[key].toString())
    override fun getShort(key: String): Short = Conversions.toShort(_map[key].toString())
    override fun getInt(key: String): Int = Conversions.toInt(_map[key].toString())
    override fun getLong(key: String): Long = Conversions.toLong(_map[key].toString())
    override fun getFloat(key: String): Float = Conversions.toFloat(_map[key].toString())
    override fun getDouble(key: String): Double = Conversions.toDouble(_map[key].toString())
    override fun getLocalDate(key: String): LocalDate = Conversions.toLocalDate(_map[key].toString())
    override fun getLocalTime(key: String): LocalTime = Conversions.toLocalTime(_map[key].toString())
    override fun getLocalDateTime(key: String): LocalDateTime = Conversions.toLocalDateTime(_map[key].toString())
    override fun getZonedDateTime(key: String): ZonedDateTime = Conversions.toZonedDateTime(_map[key].toString())
    override fun getDateTime(key: String): DateTime = Conversions.toDateTime(_map[key].toString())


    override fun get(key: String): Any? = if (_map.contains(key)) _map[key] else null
    override fun getObject(key: String): Any? = if (_map.contains(key)) _map[key] else null
    override fun containsKey(key: String): Boolean = _map.contains(key)
    override fun size(): Int = _map.size
}
